R/response.R

```diff
@@ -86,6 +86,72 @@ construct_response <- function(res){
   structure(res, class = c(class(res), "ambiorixResponse"))
 }
 
+inline_dependencies <- function(deps) {
+  deps |>
+    lapply(\(dep) {
+      if(!length(dep$src))
+        return()
+
+      if(!length(dep$src$file))
+        return()
+
+      scripts <- dep$script |>
+        lapply(\(s) {
+          content <- read_lines(file.path(dep$src$file, s)) |>
+            (\(.) paste0(., collapse = "\n"))()
+          htmltools::tags$script(type = "application/javascript", htmltools::HTML(content))
+        })
+
+      styles <- dep$stylesheet |>
+        lapply(\(s) {
+          content <- read_lines(file.path(dep$src$file, s)) |>
+            (\(.) paste0(., collapse = "\n"))()
+          htmltools::tags$style(htmltools::HTML(content))
+        })
+
+      list(scripts, styles)
+    })
+}
+
+render_htmltools <- function(x) {
+  # it it has a <body> tag we assume
+  # it's a document and render with
+  # dependencies, etc.
+  # otherwise we just render the tags.
+  q <- htmltools::tagQuery(x)
+
+  if(!length(q$closest("html")$selectedTags()))
+    return(htmltools::doRenderTags(x))
+
+  deps <- x |>
+    htmltools::findDependencies() |>
+    htmltools::resolveDependencies()
+
+  inline_deps <- inline_dependencies(deps)
+
+  xx <- htmltools::renderDependencies(deps)
+  # add <head> if not present
+  if(!length(q$find("head")$selectedTags()))
+    q$closest("html")$append(htmltools::tags$head())
+
+  # htmltools::renderDependencies(..., srcType = "href")
+  # does not work
+  rendered_deps <- htmltools::renderDependencies(deps)
+  href_deps <- grep("http", strsplit(rendered_deps, "\n")[[1]], value = TRUE)
```

Comment:
We only take href here so you should not see this in your console, can you make sure you're on the latest?

Reply:
I get it too now, boooh!

I've never had this before, the inspector triggers `GET` requests, excuse me?